From 232f00c64196b3c422283f6d4787e6b54c791be1 Mon Sep 17 00:00:00 2001
From: Rajat Srivastava <rajat.srivastava@nxp.com>
Date: Wed, 31 Oct 2018 18:14:54 +0530
Subject: [PATCH 341/706] arm64: dts: ls1028: modify qspi to fspi node in dts
 files

This patch is to move fspi node out of the soc node.

Also enable nxp flexspi driver by default.

Signed-off-by: Rajat Srivastava <rajat.srivastava@nxp.com>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-qds.dts    | 14 +++++------
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts    | 14 +++++------
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 24 +++++++++----------
 3 files changed, 26 insertions(+), 26 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
index 4c74664949a8..8dca3629f3a5 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
@@ -214,18 +214,18 @@
 	};
 };
 
-&qspi {
-	num-cs = <1>;
-	bus-num = <0>;
+&fspi {
 	status = "okay";
-
-	qflash0: s25fs512s@0 {
-		compatible = "spansion,m25p80";
+	flash0: mt35xu02g@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		spi-max-frequency = <20000000>;
+		compatible = "micron,m25p80";
 		m25p,fast-read;
+		spi-max-frequency = <20000000>;
 		reg = <0>;
+		/* The following setting enables 1-1-8 (CMD-ADDR-DATA) mode */
+		spi-rx-bus-width = <8>; /* 8 SPI Rx lines */
+		spi-tx-bus-width = <1>; /* 1 SPI Tx line */
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 107332e65dc3..5b87b0541bb7 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -160,18 +160,18 @@
 
 };
 
-&qspi {
-	num-cs = <1>;
-	bus-num = <0>;
+&fspi {
 	status = "okay";
-
-	qflash0: s25fs512s@0 {
-		compatible = "spansion,m25p80";
+	flash0: mt35xu02g@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		spi-max-frequency = <20000000>;
+		compatible = "micron,m25p80";
 		m25p,fast-read;
+		spi-max-frequency = <20000000>;
 		reg = <0>;
+		/* The following setting enables 1-1-8 (CMD-ADDR-DATA) mode */
+		spi-rx-bus-width = <8>; /* 8 SPI Rx lines */
+		spi-tx-bus-width = <1>; /* 1 SPI Tx line */
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 5b4cc3b99ada..1d063fd9f39c 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -196,6 +196,18 @@
 		};
 	};
 
+	fspi: flexspi@20c0000 {
+		compatible = "nxp,lx2160a-fspi", "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x0 0x20c0000 0x0 0x10000>,
+		    <0x0 0x20000000 0x0 0x10000000>;
+		reg-names = "FSPI", "FSPI-memory";
+		interrupts = <0 25 0x4>; /* Level high type */
+		clocks = <&clockgen 4 3>, <&clockgen 4 3>;
+		clock-names = "fspi_en", "fspi";
+	};
+
 	soc: soc {
 		compatible = "simple-bus";
 		#address-cells = <2>;
@@ -491,18 +503,6 @@
 			status = "disabled";
 		};
 
-		qspi: quadspi@20c0000 {
-			compatible = "fsl,ls2080a-qspi", "fsl,ls1088a-qspi";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0x0 0x20c0000 0x0 0x10000>,
-					<0x0 0x20000000 0x0 0x10000000>;
-			reg-names = "QuadSPI", "QuadSPI-memory";
-			interrupts = <0 25 0x4>; /* Level high type */
-			clocks = <&clockgen 4 3>, <&clockgen 4 3>;
-			clock-names = "qspi_en", "qspi";
-		};
-
 		sata: sata@3200000 {
 			compatible = "fsl,ls1028a-ahci";
 			reg = <0x0 0x3200000 0x0 0x10000>,
-- 
2.17.1

